src/app/mytools/mytools.service.ts

```diff
@@ -57,15 +61,45 @@ export class MytoolsService extends MyEntriesService<DockstoreTool, OrgToolObjec
           this.alertService.detailedError(error);
         }
       );
+
+    this.usersService
+      .userAppTools(userId)
+      .pipe(finalize(() => this.myEntriesService.setRefreshingMyEntries(false)))
+      .subscribe((appTools: Array<Workflow>) => {
+        this.workflowService.setWorkflows(appTools);
+      });
   }
+
   selectEntry(tool: DockstoreTool | Workflow | null): void {
     if (tool && tool.id) {
-      this.containersService.getContainer(tool.id, includesValidation).subscribe((result) => {
-        this.location.go('/my-tools/' + result.tool_path);
-        this.containerService.setTool(result);
-      });
+      if (MytoolsService.isAppTool(tool)) {
+        this.workflowsService.getWorkflow(tool.id, includesValidation + ',' + includesAuthors).subscribe((result) => {
+          this.location.go('/my-tools/' + result.full_workflow_path);
+          this.workflowService.setWorkflow(<AppTool>result);
+          this.containerService.setTool(null);
+        });
+      } else {
+        this.containersService.getContainer(tool.id, includesValidation).subscribe((result) => {
+          this.location.go('/my-tools/' + result.tool_path);
+          this.containerService.setTool(result);
+          this.workflowService.setWorkflow(null);
+        });
+      }
+    }
+  }
+
+  static isAppTool(tool: DockstoreTool | AppTool | OrgToolObject<DockstoreTool> | OrgWorkflowObject<Workflow>): tool is AppTool {
```

Comment:
We generally don't use static methods like this, although I know it's a hassle to inject the service. Gary didn't object, so I guess it's OK. :)